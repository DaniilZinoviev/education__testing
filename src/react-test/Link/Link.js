import React, { useState } from "react";

const STATUS = {
  HOVERED: "hover",
  NORMAL: "",
};

const Link = ({ page, children }) => {
  const [className, setClassName] = useState("test");

  const onMouseEnter = () => {
    setClassName(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setClassName(STATUS.NORMAL);
  };

  return (
    <a
      className={className}
      href={page || "#"}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
};

export default Link;
