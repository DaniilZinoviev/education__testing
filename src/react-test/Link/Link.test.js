import { describe, test, expect } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";
import Link from "./Link";

describe("component Link", () => {
  test("should test", () => {
    const component = renderer.create(
      <Link page="https://example.com/">Example</Link>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onMouseEnter();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
