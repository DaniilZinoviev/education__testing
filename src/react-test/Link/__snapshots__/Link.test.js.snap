// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Link should test 1`] = `
<a
  className="test"
  href="https://example.com/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Example
</a>
`;

exports[`component Link should test 2`] = `
<a
  className="hover"
  href="https://example.com/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Example
</a>
`;
